pg_back

```diff
@@ -268,6 +268,17 @@ else
     dumped+=( "pg_global" )
 fi
 
+# Dump configuration using SHOW ALL
+dump="$PGBK_BACKUP_DIR/pg_settings_${DUMP_DATE}.out"
+info "saving output of SHOW ALL to ${dump}"
+${PGBK_BIN}psql -X -o "${dump}" -c "SHOW ALL;" $PGBK_CONNDB
```

Comment:
I'm not fond of `SHOW ALL`. I'll suggest `pg_settings` for 2 reasons:
  * you can build a pseudo-postgresql.conf file that the DBA can put in place of the postgresql.conf file, or at the end of a default postgresql.conf file ;
  * you can filter the parameters to include.

This is tricky: PG does not start if `data_directory` is missing, or if `wal_segments` is present. The following prototype seems to work:

```bash
 psql -p12002 -Xt -c "select (case when coalesce(context,'')='internal' then '# ' else '' end) ||  name || ' = ''' || setting || ''' # ' || coalesce (sourcefile, 'initdb, pg_ctl or environment' ) from pg_settings where sourcefile is not null or source = 'override' "
```
As for `SHOW ALL`, there is no way to recover a parameter modified though ALTER ROLE SET or ALTER DATABASE. As most backups must be done with a technical superuser connected on postgres, this is not a real problem.

Another option is to be stupid, dump all pg_settings, and let the DBA sort the parameters.

Reply:
Thank you for the review. I guess I'll let the DBA sort things out from the output of show all for now